core/translate/expr.rs

```diff
@@ -1942,6 +1952,32 @@ pub fn translate_expr(
     }
 }
 
+fn allocate_registers(
```

Comment:
This should definitely be called something else than `allocate_registers()` since it translates expressions. perhaps just `translate_expr_list()`?

Reply:
not in scope for this PR, but this function addresses a common pitfall with allocating registers used as input for other instructions, and we should use it elsewhere as well.

Reply:
perhaps leave a TODO about using this in other similar places?

Reply:
I renamed it to `translate_variable_sized_function_parameter_list`. The name is quite long, but I couldn't find anything shorter that is as descriptive. Does this make sense?

I'll create a subsequent PR to refactor other functions with varying number of params to use it 👍